### PR TITLE
Filtering by Jurisdiction: Federal, State, County, and Municipality

### LINKED ISSUE #51: User input office_div filter
# Purpose: Have reps be filterable by jurisdiction: federal, state, county
Screen Shot 2021-05-22 at 6 18 28 PM

User Story: 
- Use button to filter representatives and shorten returned list of search for easier viewing

Tasks: 
1. Create buttons in campaign.vue view 
2. Have button filter by representative [levels](https://developers.google.com/civic-information/docs/v2/representatives/representativeInfoByAddress)
   a. country = **federal** 
   b. level 1 = **state**
   c. level 2 = **county** 
3. Only show representative array that corresponds 

File to change: [SearchReps](https://github.com/ProgramEquity/amplify/blob/main/src/components/SearchReps.vue) line 17

### PR DESCRIPTION
**Pull request by JamesMGreene**
Closes #51 
Closes #243 

Addresses:
- #51 

**TL;DR:** Have reps be filterable by jurisdiction: federal, state, county, and municipality

Have button filter by representative [levels](https://developers.google.com/civic-information/docs/v2/representatives/representativeInfoByAddress)
- federal = `country`
- state = `administrativeLevel1`
- county = `administrativeLevel2`
- municipality = `locality`

Only show representatives that correspond to that activated filter.

This PR builds on the filtering foundation that @mariah622 laid out in #243, then addresses the various PR Review feedback I left in that PR to reduce the overall change set.

Mariah ran into some very understandable confusion due to a logical shortcut we made by doing a `.slice(2)` on the representatives response to skip the President and VP, but once she added district-level filtering to the Google Civic API call, that lead to us omitting a _different_ first two offices instead such that it presented as data loss.

She developed an effective solution to workaround that by only slicing for federal/unspecified filtering scenarios but, starting back from the initial state, I knew we could simplify that to simplify `.filter(...)` out the offices we truly wanted to skip by name instead of changing the flow, resulting in les code changes in this PR as compared to the `main` branch. 🤷🏻‍♂️ 

---

To compare this PR to the original filtering PR #243:

👀 https://github.com/ProgramEquity/amplify/compare/filter_by_reps...filter_by_reps-revised

### CODE REVIEW

**Comment by JamesMGreene** [comment]
cc @gdomingu @paramsiddharth @mariah622 Since you're already familiar with @mariah622's PR that I based this one on, it would be much appreciated if you could give this extended PR a review. 🙏🏻 🙇🏻

**Review by mariah622 — APPROVED** [review]
These changes look great! I definitely have a better understanding of what I could have improved in my original code. This is a much cleaner cut implementation. I really appreciate the feedback and guidance 🙏

**Review by paramsiddharth — APPROVED** [review]
Looks good to me~!
